thunder/distributed/__init__.py

```diff
@@ -16,6 +20,8 @@
 
 if TYPE_CHECKING:
     from torch.distributed import ProcessGroup
+    import thunder
```

Comment:
it feels lik eL20 should be
```
from thunder.core.module import ThunderModule
```
and this should be removed since I don't think `ThunderModule` is a class residing thunder namespace as per https://github.com/Lightning-AI/lightning-thunder/blob/cb489c9532e3fdf37d684a92f13f843130a7d0a5/thunder/__init__.py#L14